test/relayserver/ServerTestEnvironments.ts

```diff
@@ -0,0 +1,227 @@
+import {
+  KeyManager,
+  RelayServer,
+  ServerDependencies,
+  TxStoreManager,
+} from '@rsksmart/rif-relay-server';
+import {
+  EnvelopingRequest,
+  EnvelopingTxRequest,
+  HubInfo,
+  RelayClient,
+  UserDefinedEnvelopingRequest,
+} from '@rsksmart/rif-relay-client';
+import { utils, Wallet } from 'ethers';
+import {
+  assertEventHub,
+  getTemporaryWorkdirs,
+  ServerWorkdirs,
+} from './ServerTestUtils';
+import {
+  RelayHubInterface,
+  RelayHub__factory,
+} from '@rsksmart/rif-relay-contracts';
+import { ethers as hardhat } from 'hardhat';
+import { expect } from 'chai';
+import { deployRelayHub } from '../TestUtils';
+import config from 'config';
+
+const provider = hardhat.provider;
+const dayInSec = 24 * 60 * 60;
+const weekInSec = dayInSec * 7;
+
+type ServerInitParams = {
+  relayOwner: Wallet;
+  serverWorkdirs?: ServerWorkdirs;
+  relayVerifierAddress: string;
+  deployVerifierAddress: string;
+  penalizerAddress?: string;
+};
+
+const initServer = async (
+  initParams: ServerInitParams,
+  getServer = getFundedServer
+) => {
+  const { relayOwner } = initParams;
+  const relayServer = await getServer(initParams);
+  await relayServer.init();
+  const latestBlock = await provider.getBlock('latest');
+  const receipts = await relayServer._worker(latestBlock.number);
+  await assertEventHub('RelayServerRegistered', receipts, relayOwner.address); // sanity check
+  await assertEventHub('RelayWorkersAdded', receipts, relayOwner.address); // sanity check
+  await relayServer._worker(latestBlock.number + 1);
+
+  return relayServer;
+};
+
+const getFundedServer = async (
+  initParams: ServerInitParams,
+  getServer = getServerInstance,
+  unstakeDelay = weekInSec
+) => {
+  const relayServer = await getServer(initParams);
+
+  const { relayOwner } = initParams;
+
+  const { relayHubAddress, relayManagerAddress } = relayServer.pingHandler();
+
+  await relayOwner.sendTransaction({
+    value: utils.parseEther('2'),
+  });
+
+  const relayHub = RelayHub__factory.connect(relayHubAddress, provider);
+
+  await relayHub
+    .connect(relayOwner)
+    .stakeForAddress(relayManagerAddress, unstakeDelay, {
+      value: utils.parseEther('1'),
+    });
+
+  return relayServer;
+};
+
+const getServerInstance = async ({
+  serverWorkdirs,
+  deployVerifierAddress,
+  relayVerifierAddress,
+  penalizerAddress,
+}: ServerInitParams) => {
+  const { workdir } = getTemporaryWorkdirs();
+
+  const managerKeyManager = createKeyManager(serverWorkdirs?.managerWorkdir);
+  const workersKeyManager = createKeyManager(serverWorkdirs?.workersWorkdir);
+  const txStoreManager = new TxStoreManager({
+    workdir: serverWorkdirs?.workdir ?? workdir,
+  });
+
+  const dependencies: ServerDependencies = {
+    txStoreManager,
+    managerKeyManager,
+    workersKeyManager,
+  };
+
+  const relayHub = await deployRelayHub(penalizerAddress, {});
```

Comment:
Sorry, but apart from the configuration, I don't see any difference between the hub and the verifiers. Could you please elaborate a bit more?

Reply:
The verifiers are deployed outside this function and the address are passed, therefore the responsibility of creating verifiers its from the one that is going to use this function. Right now, the responsibility of deploying the relayHub its from this function. Take in consideration that we can get the relayHubAddress at any time from the server but we cannot get the verifiers, therefore I decided to not deploy it here, since we will not have access to it.